### ISSUE TITLE
Camera inverts when device is rotated

### ISSUE THREAD

**Issue by miguelpruivo** [rmtmckenzie/flutter_qr_mobile_vision#84]
For some reason, whenever the device is rotated while the camera is opened, it does some weird invert in runtime. Looks like it is flipping for no reason.

To replicate it just open it with your device and rotate it while scanning.

**Comment by rmtmckenzie** [comment]
The plugin does have camera rotation built in for if the device is rotated, but it should only do that if the actual app window rotates (i.e. if you have screen rotation turned off for your device, it shouldn't rotate the camera view by default). Without the rotation I perform, the actual camera view would be rotated wrong when the overall window rotates.

Could you tell me which device & OS version you're seeing this with, whether you have auto-rotate turned on, and/or whether you've done anything in your app like disabling rotation (i.e. by setting the app to portrait only).

**Comment by miguelpruivo** [comment]
Hi @rmtmckenzie. The app is set to work on portrait mode only (both in iOS and Android) and the camera still rotates. 

Besides that, and even if the device orientation is locked (on the device's settings) it will rotate anyway. It's happening with all devices I've tested (some Android and some iPhone). Looks like a general issue.

**Comment by rmtmckenzie** [comment]
Hmm. That's weird. There were some updates to the rotation plugin that were supposed to make it able to work when rotation is disabled, but I thought that was turned off by default. I'll try to take a look at that soon.

**Comment by miguelpruivo** [comment]
I don't know if you saw my edited answer @rmtmckenzie but it's only happening with iOS. Tested with an Android and it seems to be working fine.

**Comment by miguelpruivo** [comment]
@rmtmckenzie after further investigation I've found that the issue is actually coming from your other plugin (orientation), more specifically from [here](https://github.com/rmtmckenzie/flutter_native_device_orientation/blob/master/ios/Classes/OrientationListener.m#L40) where you are getting the device orientation instead of interface orientation. Here, if you use something like
```
UIInterfaceOrientation interfaceOrientation = [[UIApplication sharedApplication] statusBarOrientation];
```
you'd get the current's device orientation respecting the rotation lock whereas with what you have you'll *always* get the device orientation even though the interface is not being rotated (when locked). 

My suggestion would be to add an helper method to your orientation plugin that checks if the rotation is locked or not and then use it accordingly in your qr code reader package somewhere [here](https://github.com/rmtmckenzie/flutter_qr_mobile_vision/blob/master/lib/qr_camera.dart#L199)  by checking it before actually rotating the `SizedBox` to prevent this issue. 

Let me know if you want me to issue a PR for this as I really would rather to have my production app to use the dart pub library instead of a forked one.

**Comment by Baileypollard** [comment]
@miguelpruivo What was your solution to this issue?

**Comment by miguelpruivo** [comment]
@Baileypollard I forked it and fixed it, you can use it directly from [here](https://github.com/miguelpruivo/flutter_qr_mobile_vision) if you want.

**Comment by miguelpruivo** [comment]
@Baileypollard I forked it and added a workaround to it, you can use it directly from [here](https://github.com/miguelpruivo/flutter_qr_mobile_vision) if you want.

**Comment by Baileypollard** [comment]
@miguelpruivo Thanks for that. Have you experienced low Frames/s on IOS devices specifically?

**Comment by miguelpruivo** [comment]
@Baileypollard yes. I ended up using the [fast_qr_reader_view](https://github.com/miguelpruivo/fast_qr_reader_view) instead, also forked because it had some issue by the time. 

I have a production app that relies on QR code scanning and this was one of the few solutions that worked, after some researching and trying with a lot of different reader packages.

**Comment by Baileypollard** [comment]
@miguelpruivo Have you ran into any issues where fast_qr_reader_view isn't scanning on IOS?

**Comment by miguelpruivo** [comment]
@Baileypollard not actually, and it has been in production for a while now (since June). But, I'm only using it to scan QR Codes, I don't know if it has any other issue with different kind of codes.

**Comment by MikeSWelch** [comment]
I have forked and fixed the problem: https://github.com/MikeSWelch/flutter_qr_mobile_vision

I have also submitted a pull request.

**Comment by rmtmckenzie** [comment]
I think this is fixed in the newest version, 1.0.0.
